src/lib.rs

```diff
@@ -184,29 +189,43 @@ unsafe extern "C" fn init_function(
         let s = core::slice::from_raw_parts(ptr, len); // Explicit nul skip.
 
         if !s.is_empty() {
-            v.push(s.to_owned());
+            v.push(Box::leak(s.into()));
         }
 
         // Safety: This will never wrap and after incrementing
         // past the last array element, the loop will stop.
         applep = applep.add(1);
     }
 
+    let vslice = v.leak::<'static>();
     // `Relaxed` is fine because the store of `data` with
     // `Release` acts as a fence, and `len` is always loaded
     // after `data`.
-    ARGS_LEN.store(v.len(), Ordering::Relaxed);
-    ARGS_DATA.store(
-        Box::into_raw(v.into_boxed_slice()).cast::<Vec<u8>>(),
-        Ordering::Release,
-    );
+    ARGS_LEN.store(vslice.len(), Ordering::Relaxed);
+    ARGS_DATA.store(vslice.as_mut_ptr(), Ordering::Release);
+    // Disarm the abort guard.
+    core::mem::forget(panic_in_static_ctor_sounds_bad);
 }
 
 extern "C" {
     /// Provided by libc or compiler_builtins.
     fn strlen(s: *const c_char) -> usize;
 }
 
+struct AbortGuard;
```

Comment:
This is unrelated but a good idea (I didn't think of it when you asked me to look before).